### PR TITLE
Patch for quickopen

### PR DESCRIPTION
**Pull request by rothfels**
- enable multi-workspace extension host multiplexing for search requests
- rewrite `fsPath` construction logic for our `git://` based document URIs; these are used by VS code for quickopen entry labels

### CODE REVIEW

File: src/vs/base/common/uri.ts [hunk 1]
```diff
@@ -134,7 +134,10 @@ export default class URI {
 				value = this._path[1].toLowerCase() + this._path.substr(2);
 			} else {
 				// other path
-				value = this._path;
+				value = this._path.replace(/^\//, "")
```
Comment (nicksnyder) [inline src/vs/base/common/uri.ts:137]: A comment here to explain what is happening and why would be nice (like the other cases)

File: src/vs/workbench/parts/search/common/search.ts [hunk 2]
```diff
@@ -53,16 +69,20 @@ export namespace WorkspaceSymbolProviderRegistry {
 		};
 	}
 
-	export function all(): IWorkspaceSymbolProvider[] {
-		return _supports.slice(0);
+	export function all(workspace?: IWorkspace): IWorkspaceSymbolProvider[] {
+		const supports = _supports.slice(0);
```
Comment (nicksnyder) [inline src/vs/workbench/parts/search/common/search.ts:73]: This is only to create a copy right? The filter case already creates a copy so you could do this

```
if (workspace) {
    return _supports.filter(support => support.workspace && support.workspace.resource && support.workspace.resource.toString() === workspace.resource.toString())
}
return _supports.slice(0); // create a copy
```

**Review by nicksnyder — APPROVED** [review]
Makes sense

**Comment by rothfels** [comment]
Closing in favor of https://github.com/sourcegraph/vscode/pull/6

**Pull request closed without merging**